### ISSUE TITLE
vscode debian package misbehaves with newer recommended apt files structure

### ISSUE THREAD

**Issue by AndreaBorgia-Abo** [microsoft/vscode#204014]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.86.0-1706698139
- OS Version: Debian/testing

Steps to Reproduce:

1. have previous version already installed
2. migrate the (deprecated) /etc/apt/trusted* key folders to /etc/apt/keyrings/* and rewrite all ".list" files with the new deb822 format ".sources"
3. apt update
4. apt dist-upgrade
5. try again

After step 4 apt complains about a missing directory:
```
Configurazione di code (1.86.0-1706698139)...
/var/lib/dpkg/info/code.postinst: riga 77: /etc/apt/trusted.gpg.d/microsoft.gpg: File o directory non esistente
```

After step 5 it will have created a functional duplicate of /etc/apt/sources.list.d/vscode.sources as vscode.list;
this is somehow related to #189274 in that the root cause is always the same, main difference is that the key file itself will apparently not be duplicated and apt won't complain (too much).

**Comment by AndreaBorgia-Abo** [comment]
FYI a reasonable workaround is creating an empty /etc/apt/sources.list.d/vscode.list file, then making it mode 444 and immutable. No error is reported on upgrade and no unwanted changes to apt sources occur.

**Comment by rzhao271** [comment]
Closing in favour of https://github.com/microsoft/vscode/issues/215745.
